blogs/2021/01/ext-bisect.md

```diff
@@ -0,0 +1,40 @@
+
+Because there are no bad extensions, we have created one: [Extension Bisect Demo](https://marketplace.visualstudio.com/items?itemName=jrieken.bisectdemo). When installed, it will annoyingly reset your cursor whenever you reach the word “bisect”. Use that extension to follow along.
+
+Finding a bad extension is easy: Open the extension view, [disable an extension]( https://code.visualstudio.com/docs/editor/extension-gallery#_disable-an-extension), reload, and check for the issue. If the problem is gone that extension is bad and you are done. Otherwise, re-enable the extension and repeat the process with the next extension.
+
+![Disabling an extension step by step](./disable_manually.png)
+
+If you are lucky the first extension is bad, if you are unlucky it’s the last extension. Using computer science language this means with `N` extensions you have a worst-case of `O(N)`, and an average-case of `O(N/2)`. Because this algorithm is operated by a human even small values of `N` are laborious and mindless. This is where extension bisect comes into play. It is much better in the worst- and average-case because it disables extensions by halves. This is like the problem of finding out which commit in git has caused a regression. Git has the [`git bisect`](https://git-scm.com/docs/git-bisect)-command for that and that inspired us to do the same for extensions.
+
+Let’s use a sample:  I have 24 extensions installed and the 8th extension is bad. We know that the iterative approach requires 8 steps. What about bisect?
+
+![The extension bisect process](bisect.gif)
+
+1. Bisect divides 24 extensions into two halves of 12 and 12 extensions and it disables all 12 extensions of the second half.
+2. In this sample the 8th extension is bad, it is the first half and not disabled. Things are still bad. Because of that bisect divides the first 12 extensions into 6 enabled and 6 disabled extensions. All other extensions are also re-enabled.
+3. The 8th extension is now disabled. Things are good now. That means bisect proceeds on the second half (extensions 6-11) and divides them into 3 enabled and 3 disabled extensions.
+4. Now, the 8th extension is re-enabled, and things have turned bad. This means bisect proceeds on the first half. It divides them into 1 enabled and 2 disabled extensions.
+5. The 8th extension is now disabled, things are good again and bisect proceeds on the second half, dividing it into 1 enabled and 1 disabled extension.
+6. The 8th extension is the only disabled extension and things are good. This means we have found the bad extension and that we are done.
+
```

Comment:
Some extensions require a reload and some do not. Since we're disabling multiple extensions at one time, is a full editor reload always desired?